src/main/java/com/zendesk/maxwell/bootstrap/MaxwellBootstrap.java

```diff
@@ -0,0 +1,155 @@
+package com.zendesk.maxwell.bootstrap;
+
+import com.zendesk.maxwell.MaxwellLogging;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import snaq.db.ConnectionPool;
+
+import java.io.Console;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class MaxwellBootstrap {
+	static final Logger LOGGER = LoggerFactory.getLogger(MaxwellBootstrap.class);
+
+	private static final long UPDATE_PERIOD_MILLIS = 250;
+	private static final long DISPLAY_PROGRESS_WARMUP_MILLIS = 5000;
+	private static Console console = System.console();
+
+	private void run(String[] argv) throws Exception {
+		MaxwellBootstrapConfig config = new MaxwellBootstrapConfig(argv);
+		if ( config.log_level != null ) {
+			MaxwellLogging.setLevel(config.log_level);
+		}
+		ConnectionPool connectionPool = getConnectionPool(config);
+		try ( Connection connection = connectionPool.getConnection() ) {
+			int rowCount = getRowCount(connection, config);
+			long rowId = insertBootstrapStartRow(connection, config);
+			boolean isComplete = false;
+			int insertedRowsCount = 0;
+			Long startedTimeMillis = null;
+			while ( !isComplete ) {
+				if ( insertedRowsCount < rowCount ) {
```

Comment:
getRowCount() may return a different number of rows than the actual bootstrap is seeing. Consider the following scenario:

1. 1 million records in table.
2. getRowCount() returns 1 million
3. insertBootstrapStartRow()
4. 1 million rows deleted by outside process
5. Maxwell sees the bootstrap start row.
6. Maxwell does a "select *" and inserts 0 rows into the stream.
7. Maxwell marks the bootstrap row as completed.

insertedRowsCount will always be 0. This while loop will never set isComplete and will block forever.